core/network/src/protocols/transmitter/protocol.rs

```diff
@@ -0,0 +1,100 @@
+use std::time::Instant;
+
+use futures::channel::mpsc::UnboundedSender;
+use protocol::Bytes;
+use tentacle::context::ProtocolContextMutRef;
+use tentacle::traits::SessionProtocol;
+
+use super::message::{ReceivedMessage, SeqChunkMessage};
+use super::{DATA_SEQ_TIMEOUT, MAX_CHUNK_SIZE};
+
+pub struct TransmitterProtocol {
+    data_tx:            UnboundedSender<ReceivedMessage>,
+    data_buf:           Vec<u8>,
+    current_data_seq:   u64,
+    first_seq_bytes_at: Instant,
+}
+
+impl TransmitterProtocol {
+    pub fn new(data_tx: UnboundedSender<ReceivedMessage>) -> Self {
+        TransmitterProtocol {
+            data_tx,
+            data_buf: Vec::new(),
+            current_data_seq: 0,
+            first_seq_bytes_at: Instant::now(),
+        }
+    }
+}
+
+impl SessionProtocol for TransmitterProtocol {
+    fn received(&mut self, ctx: ProtocolContextMutRef, data: Bytes) {
+        let peer_id = match ctx.session.remote_pubkey.as_ref() {
+            Some(pk) => pk.peer_id(),
+            None => {
+                // Dont care result here, connection/keeper will also handle this.
+                let _ = ctx.disconnect(ctx.session.id);
+                return;
+            }
+        };
+        let session_id = ctx.session.id;
+
+        if data.len() < 10 {
```

Comment:
10 = Minimum protocol package?

Reply:
```seq``` takes 8 bytes, ```bool``` takes 1 byte, so a valid data length must be bigger or equal than 10.